src/utils.rs

```diff
@@ -153,13 +235,31 @@ pub fn apply_TextEdits(lines: &[String], edits: &[TextEdit]) -> Fallible<Vec<Str
     }
 
     let mut text = lines.join("\n");
+    let mut offset = position_to_offset(&lines, &position);
     for (start, end, new_text) in edits_by_index {
         let start = std::cmp::min(start, text.len());
         let end = std::cmp::min(end, text.len());
         text = String::new() + &text[..start] + new_text + &text[end..];
+
+        // Update offset only if the edit's start and end are both before it.
+        // Edits after the offset do not affect it.
+        // Edits whose range contains the offset cause unpredictable effect.
+        if end < offset {
+            offset += new_text.len();
```

Comment:
This may have issue if the new_text uses "\r\n" as new line which will be counted as two characters. I will test if any adjustment is needed here.